app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt

```diff
@@ -183,7 +183,10 @@ class HistoryFragment : Fragment(), CoroutineScope by MainScope(), BackHandler {
         else -> super.onOptionsItemSelected(item)
     }
 
-    override fun onBackPressed(): Boolean = (historyComponent.uiView as HistoryUIView).onBackPressed()
+    override fun onBackPressed(): Boolean {
+        emitChange { HistoryChange.ExitEditMode }
```

Comment:
While this appears to resolve the problem, we wind up with more events than we need.

Changing HistoryUIView:139 to emit BackPressed instead of SwitchMode accomplishes the same thing with fewer events.

Reply:
Understood. I'll update the pull request.  Thanks,